src/Microsoft.AspNetCore.OData/Extensions/ODataServiceCollectionExtensions.cs

```diff
@@ -22,11 +22,14 @@ namespace Microsoft.AspNet.OData.Extensions
     public static class ODataServiceCollectionExtensions
     {
         /// <summary>
-        /// Adds essential OData services to the specified <see cref="IServiceCollection" />.
+        /// Adds core OData services to the specified <see cref="IServiceCollection" />.
         /// </summary>
+        /// <remarks>
+        /// Does not add OData formatters or the OData action selector.
+        /// </remarks>
         /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
         /// <returns>An <see cref="IODataBuilder"/> that can be used to further configure the OData services.</returns>
-        public static IODataBuilder AddOData(this IServiceCollection services)
+        public static IODataBuilder AddODataCore(this IServiceCollection services)
```

Comment:
I understand the desire for this change but I'm hoping to get containers in the pre-request container and out of the global container; as such, I'm not sure we'll need two function long term.

Reply:
You mean get rid of `IPerRequestContainer` entirely? Sure, this won't be necessary then, but wouldn't it be a bit heavy on resources to create the whole OData container for each request?

Reply:
Nope, I was thinking that we'd keep the per-request container but that code in what is now AddOData() would wind up inside the per-request container, which would wind up eliminating the need for the now AddOData() and leaving only AddODataCore(), at which point AddOData() would probably be a better name.

For now, I'm not pushing back on the change, just wanted to throw that out and get your thoughts on this. If we add AddODataCore() now, it will be hard to get rid of later.

Reply:
I see. Should I delete this from the PR in the meantime?

Reply:
How about we leave it and mark it as deprecated? If it winds up staying, we can pull the attribute out but it's a warning that it might go away.

[Obsolete("Use to avoid adding formatters and action selector; may not be needed in the future.")]